include/cutlass/arch/arch.h

```diff
@@ -47,20 +47,24 @@ namespace arch {
 CUTLASS_DEVICE
 int LaneId() {
   int ret;
-#if !defined(CUTLASS_ENABLE_SYCL) || defined(__SYCL_CUDA_ARCH__)
+#if defined(__CUDA_ARCH__) || defined(__SYCL_CUDA_ARCH__)
   asm ("mov.u32 %0, %%laneid;" : "=r"(ret) : );
-#endif
   return ret;
+#else
+  CUTE_GCC_UNREACHABLE;
```

Comment:
Are you missing an assert here?

Reply:
That's a good point... I had previously run a small test to see what happens if a `__builtin_unreachable()` is reached, and it aborted with a relevant error, so I figured that was sufficient. However, technically reaching a `__builtin_unreachable` is [UB](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html#index-_005f_005fbuiltin_005funreachable). Perhaps we could add `assert(false);` to the definition of `CUTE_GCC_UNREACHABLE` and `CUTLASS_GCC_UNREACHABLE`?

Reply:
Done :+1: